File: java/samples/search/src/main/java/com/myscript/iink/samples/search/JiixRawContent.java
/*
 * Copyright (c) MyScript. All rights reserved.
 */

package com.myscript.iink.samples.search;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class JiixRawContent
{
    @SerializedName("type")
    public String type;

    @SerializedName("elements")
    public List<Element> elements;

    public class Element
    {
        @SerializedName("type")
        public String type;

        @SerializedName("words")
        public List<Word> words;

        @SerializedName("chars")
        public List<Char> chars;
    }

    public class Word
    {
        @SerializedName("label")
        public String label;

        @SerializedName("first-char")
        public int firstChar;

        @SerializedName("last-char")
        public int lastChar;

        @SerializedName("bounding-box")
        public BoundingBox boundingBox;
    }

    public class Char
    {
        @SerializedName("label")
        public String label;

        @SerializedName("word")
        public int word;

        @SerializedName("bounding-box")
        public BoundingBox boundingBox;
    }

    public class BoundingBox
    {
        @SerializedName("x")
        public float x;

        @SerializedName("y")
        public float y;

        @SerializedName("width")
        public float width;

        @SerializedName("height")
        public float height;
    }
}

File: java/samples/search-kt/src/main/java/com/myscript/iink/samples/search/JiixRawContent.kt
/*
 * Copyright (c) MyScript. All rights reserved.
 */

package com.myscript.iink.samples.search

class JiixRawContent {
    @com.google.gson.annotations.SerializedName("type")
    var type: String? = null

    @com.google.gson.annotations.SerializedName("elements")
    var elements: List<Element>? = null

    @com.google.gson.annotations.SerializedName("children")
    var textDocumentElements: List<Element>? = null

    inner class Element {
        @com.google.gson.annotations.SerializedName("type")
        var type: String? = null

        @com.google.gson.annotations.SerializedName("words")
        var words: List<Word>? = null

        @com.google.gson.annotations.SerializedName("chars")
        var chars: List<Char>? = null
    }

    inner class Word {
        @com.google.gson.annotations.SerializedName("label")
        var label: String? = null

        @com.google.gson.annotations.SerializedName("first-char")
        var firstChar = 0

        @com.google.gson.annotations.SerializedName("last-char")
        var lastChar = 0

        @com.google.gson.annotations.SerializedName("bounding-box")
        var boundingBox: BoundingBox? = null
    }

    inner class Char {
        @com.google.gson.annotations.SerializedName("label")
        var label: String? = null

        @com.google.gson.annotations.SerializedName("word")
        var word = 0

        @com.google.gson.annotations.SerializedName("bounding-box")
        var boundingBox: BoundingBox? = null
    }

    inner class BoundingBox {
        @com.google.gson.annotations.SerializedName("x")
        var x = 0f

        @com.google.gson.annotations.SerializedName("y")
        var y = 0f

        @com.google.gson.annotations.SerializedName("width")
        var width = 0f

        @com.google.gson.annotations.SerializedName("height")
        var height = 0f
    }
}
